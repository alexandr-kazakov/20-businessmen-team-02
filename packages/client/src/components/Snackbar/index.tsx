import React, { useEffect } from 'react'
import { useAppDispatch, useAppSelector } from '../../app/redux/hooks'
import { hideSnackBar } from './redux/snackbarSlice'

import styles from './styles.module.scss'

export const Snackbar: React.FC = () => {
  const { message, isShow } = useAppSelector(state => state.snack)
  const dispatch = useAppDispatch()

  const handleClose = () => {
    dispatch(hideSnackBar())
  }

  useEffect(() => {
    let timer: ReturnType<typeof setTimeout> | undefined
    if (isShow) {
      timer = setTimeout(() => {
        dispatch(hideSnackBar())
      }, 3000)
    }
    return () => {
      timer && clearTimeout(timer)
    }
  }, [isShow, dispatch])

  return isShow ? (
    <div className={styles.container}>
      <div className={styles.label}>{message}</div>
      <div className={styles.close} onClick={handleClose}>
        &times;
      </div>
    </div>
  ) : null
}
