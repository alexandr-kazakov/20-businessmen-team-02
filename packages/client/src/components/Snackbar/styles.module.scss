.container {
  position: fixed;
  z-index: 1000;
  bottom: 1rem;
  left: 50%;
  transform: translateX(-50%);
  height: auto;
  padding: 0.625rem 1rem;
  border-radius: 0.75rem;
  border: transparent;
  background-color: hsl(1.74, 100%, 65%);
  color: white;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);

  display: flex;
  justify-content: center;
  align-items: center;
  
  @keyframes slide-up {
    from {
      bottom: 0;
      opacity: 0;
    }
    to {
      bottom: 1.5rem;
      opacity: 1;
    }
  }
  animation: 500ms ease-out forwards slide-up;
}

.close {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 0.875rem;
  padding: 0;
  margin-left: 1rem;
  height: 1.75rem;
  width: 1.75rem;
  text-align: center;
  border: none;
  border-radius: 50%;
  background-color: transparent;
  color: white;
  cursor: pointer;

  &:hover {
    background-color: hsl(1.74, 100%, 60%);
  }
}
