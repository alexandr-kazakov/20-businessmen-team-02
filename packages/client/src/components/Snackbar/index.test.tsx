import React from 'react'
import { render } from '@testing-library/react'
import { Snackbar } from '@/components/Snackbar/index'

jest.mock('react-redux', () => ({
  useSelector: () => ({ message: 'error', isShow: true }),
  useDispatch: jest.fn()
}));

test('should render Snackbar component', () => {
  const {asFragment} = render(<Snackbar/>)

  expect(asFragment()).toMatchSnapshot()
})