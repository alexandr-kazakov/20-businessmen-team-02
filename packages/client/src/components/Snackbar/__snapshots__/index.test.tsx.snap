// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Snackbar component 1`] = `
<DocumentFragment>
  <div>
    <div>
      error
    </div>
    <div>
      Ã—
    </div>
  </div>
</DocumentFragment>
`;
